raytrace.py

```diff
@@ -583,7 +729,16 @@ def next_direction(self, face):
 
     def update_energy(self):
         # TODO: @Ramon
-        pass
+        point_1 = self.points[-1]
+        point_2 = self.points[-2]
+        solid_1 = self.scene.solid_at_point(point_1)
+        solid_2 = self.scene.solid_at_point(point_2)
+        if solid_1 and solid_1 == solid_2:
```

Comment:
I'd suggest just to look at the midpoint of point_1 and point_2. point_1 and point_2 are on the faces and the solid_at_point method could fail. Also, this way you only have one material to look at.